evalml/tests/component_tests/test_exponential_smoothing_regressor.py

```diff
@@ -0,0 +1,229 @@
+    train_features_index_dt,
+    train_target_index_dt,
+    train_none,
+    datetime_feature,
+    no_features,
+    get_X_y,
+):
+    X_train, _, y_train = get_X_y(
+        train_features_index_dt,
+        train_target_index_dt,
+        train_none,
+        datetime_feature,
+        no_features,
+        test_features_index_dt=False,
+    )
+
+    if not train_none:
+        if train_features_index_dt:
+            assert isinstance(X_train.index, pd.DatetimeIndex)
+        else:
+            assert not isinstance(X_train.index, pd.DatetimeIndex)
+        if datetime_feature:
+            assert X_train.select_dtypes(include=["datetime64"]).shape[1] == 1
+    if train_target_index_dt:
+        assert isinstance(y_train.index, pd.DatetimeIndex)
+    else:
+        assert not isinstance(y_train.index, pd.DatetimeIndex)
+
+    clf = ExponentialSmoothingRegressor()
+    X_train_no_dt = clf._remove_datetime(X_train, features=True)
+    y_train_no_dt = clf._remove_datetime(y_train)
+
+    if train_none:
+        assert X_train_no_dt is None
+    else:
+        assert not isinstance(X_train_no_dt.index, pd.DatetimeIndex)
+        if no_features:
+            assert X_train_no_dt.shape[1] == 0
+        if datetime_feature:
+            assert X_train_no_dt.select_dtypes(include=["datetime64"]).shape[1] == 0
+
+    assert not isinstance(y_train_no_dt.index, pd.DatetimeIndex)
+
+
+def test_match_indices(get_X_y):
+    X_train, _, y_train = get_X_y(
+        train_features_index_dt=False,
+        train_target_index_dt=False,
+        train_none=False,
+        datetime_feature=False,
+        no_features=False,
+        test_features_index_dt=False,
+    )
+
+    assert not X_train.index.equals(y_train.index)
+
+    clf = ExponentialSmoothingRegressor()
+    X_, y_ = clf._match_indices(X_train, y_train)
+    assert X_.index.equals(y_.index)
+
+
+def test_set_forecast(get_X_y):
+    from sktime.forecasting.base import ForecastingHorizon
+
+    _, X_test, _ = get_X_y(
+        train_features_index_dt=False,
+        train_target_index_dt=False,
+        train_none=False,
+        datetime_feature=False,
+        no_features=False,
+        test_features_index_dt=False,
+    )
+
+    clf = ExponentialSmoothingRegressor()
+    fh_ = clf._set_forecast(X_test)
+    assert isinstance(fh_, ForecastingHorizon)
+    assert len(fh_) == len(X_test)
+    assert fh_.is_relative
+
+
+def test_feature_importance(ts_data):
+    X, y = ts_data
+    clf = ExponentialSmoothingRegressor()
+    with patch.object(clf, "_component_obj"):
+        clf.fit(X, y)
+        assert clf.feature_importance == np.zeros(1)
+
+
+@pytest.mark.parametrize(
+    "train_none, train_features_index_dt, "
+    "train_target_index_dt, no_features, "
+    "datetime_feature, test_features_index_dt",
+    [
+        (True, False, False, False, False, False),
+        (False, True, True, False, False, True),
+        (False, True, True, False, False, False),
+    ],
+)
+def test_fit_predict(
```

Comment:
For this test, seems like `no_features` and `datetime_feature` are always false in the 3 parameterizations. Should these be tested for this, since this is testing that fit and predict perform as we'd expect?